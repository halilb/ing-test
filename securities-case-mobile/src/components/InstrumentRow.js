import React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

import PriceDiff from './PriceDiff';
import { commonStyles } from '../helpers/theme';

const InstrumentRow = ({ code, price, priceDiffPercentage }) => (
  <View style={styles.container}>
    <Text style={styles.code}>{code}</Text>
    <View style={styles.priceContainer}>
      <Text style={styles.price}>{price}</Text>
      <PriceDiff diff={priceDiffPercentage} />
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: 0,
    paddingVertical: 8,
  },
  code: {
    flex: 1,
    fontWeight: 'bold',
  },
  priceContainer: {
    flex: 0,
    flexDirection: 'row',
    alignItems: 'center',
  },
  price: {
    color: '#B8B8B8',
    marginRight: 8,
  },
});

InstrumentRow.propTypes = {
  code: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  priceDiffPercentage: PropTypes.number.isRequired,
};

export default InstrumentRow;
