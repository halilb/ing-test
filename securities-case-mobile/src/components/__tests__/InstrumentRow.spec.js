import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import InstrumentRow from '../InstrumentRow';

it('renders correctly', () => {
  const snapshot = renderer
    .create(
      <InstrumentRow code={'Ing'} price={10.44} priceDiffPercentage={-0.41} />
    )
    .toJSON();
  expect(snapshot).toMatchSnapshot();
});
