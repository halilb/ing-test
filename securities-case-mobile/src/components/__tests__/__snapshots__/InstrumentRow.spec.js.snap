// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 0,
      "paddingVertical": 8,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "flex": 1,
        "fontWeight": "bold",
      }
    }
  >
    Ing
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 0,
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#B8B8B8",
          "marginRight": 8,
        }
      }
    >
      10.44
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 2,
            "paddingVertical": 2,
            "width": 52,
          },
          Object {
            "backgroundColor": "#F1374A",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 12,
          }
        }
      >
        -0.41
        %
      </Text>
    </View>
  </View>
</View>
`;
